pcdsdevices/pseudopos.py

```diff
@@ -1037,16 +1033,83 @@ def inverse(self, real_pos: tuple) -> tuple:
-        pseudo_field, = self.PseudoPosition._fields
-        real_field, = self.RealPosition._fields
+        pseudo_field, real_field = self._get_field_names()
+        xp, fp = self._load_table_arrays(x_name=real_field, f_name=pseudo_field)
 
         pseudo_value = np.interp(
             values[real_field],
-            self._table_data_by_name[real_field],
-            self._table_data_by_name[pseudo_field]
+            xp,
+            fp,
         )
         return self.PseudoPosition(**{pseudo_field: pseudo_value})
 
+    def _get_field_names(self) -> tuple[str, str]:
+        """
+        Returns the name of the pseudo field and the name of the real field.
+
+        Returns
+        -------
+        fields: tuple of str
+            (pseudo_field, real_field)
+        """
+        pseudo_field, = self.PseudoPosition._fields
+        real_field, = self.RealPosition._fields
+
+        return pseudo_field, real_field
+
+    def _load_table_arrays(self, x_name: str, f_name: str) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Load array data from the lookup table in a format ready for np.interp.
+
+        This returns the xp and fp arguments that np.interp is expecting.
+        xp is the x-coordinates of the data points, and
+        fp is the y-coordinates of the data points.
+
+        np.interp is expecting xp to be strictly increasing.
+        If xp is not strictly increasing, this function will try reversing
+        the array ordering to make it so, which will work if xp is strictly
+        decreasing.
+
+        If xp is neither strictly increasing nor strictly decreasing
+        and therefore cannot be coerced into a strictly increasing sequence,
+        this will show a warning but continue anyway.
+
+        Parameters
+        ----------
+        x_name: str
+            The name associated with the x-axis data.
+        f_name: str
+            The name associated with the y-axis data.
+
+        Returns
+        xp, fp: tuple of np.ndarray
+            The xp and fp arguments needed for np.interp.
+        """
+        xp = self._table_data_by_name[x_name]
+        fp = self._table_data_by_name[f_name]
+
+        # xp must be strictly increasing for np.interp
+        if not is_strictly_increasing(xp):
+            # Try reverse
+            xp = xp[::-1]
+            fp = fp[::-1]
+            # Check one more time in case neither direction works
+            if not is_strictly_increasing(xp):
+                self.log.warning(
+                    "Lookup table is not strictly increasing or decreasing! "
+                    "This will give inconsistent results!"
+                )
+
+        return xp, fp
+
+
+def is_strictly_increasing(arr: np.ndarray) -> bool:
+    """
+    Returns True if axis 1 of arr is strictly increasing and False otherwise.
+    """
+    # see numpy.interp docs https://numpy.org/doc/stable/reference/generated/numpy.interp.html
+    return np.all(np.diff(arr) > 0)
```

Comment:
I guess monotonically increasing does mean strictly increasing (never decreasing or staying the same).  We just had the wrong idea of "monotonically increasing" before.

Reply:
I actually re-read the docs and found that they specified _strictly_ increasing instead of _monotonically_ increasing, the difference being that a monotonic sequence is allowed to repeat adjacent numbers or have a zero slope.